app/scripts/controllers/product/EditLoanProductController.js

```diff
@@ -241,8 +245,71 @@
                 scope.formData.minimumGap = scope.product.minimumGap;
                 scope.formData.maximumGap = scope.product.maximumGap;
                 scope.formData.canUseForTopup = scope.product.canUseForTopup;
+
+                //Rate Module
+                scope.formData.rates = scope.product.rates;
+                scope.rateOptions = scope.product.rateOptions || [];
+                scope.calculatedRatePerPeriod = scope.product.interestRatePerPeriod;
+
+                if (scope.formData.rates && scope.formData.rates.length>0){
+                    scope.rateFlag=true;
+                }
+                scope.formData.rates.forEach(function(rate){
+                    scope.rateOptions.forEach(function(rateOption, index, array){
+                        if(rate.name === rateOption.name){
+                            scope.rateOptions.splice(index,1);
+                        }
+                    });
+                });
+
             });
 
+            //Rate
+            scope.rateSelected = function(currentRate){
+
+                if(currentRate){
+                    scope.formData.rates.push(currentRate);
+                    scope.rateOptions.splice(scope.rateOptions.indexOf(currentRate),1);
+                    scope.currentRate = '';
+                    scope.calculateRates();
+                }
+            };
+
+            scope.calculateRates = function(){
+                var total = 0;
+                var minRate = 0;
+                scope.formData.rates.forEach(function(rate){
+                    if(rate.percentage < minRate || minRate === 0){
+                        minRate = rate.percentage;
+                    }
+                    total += rate.percentage;
+                });
+
+                if (minRate===0){
+                  minRate=undefined;
+                }
+                if (total===0){
+                  total=undefined;
+                  scope.rateFlag=false;
+                }
+                scope.formData.minInterestRatePerPeriod = minRate;
+
+                //Assign the same total range to this values.
+                scope.formData.interestRatePerPeriod = total;
+                scope.formData.maxInterestRatePerPeriod = total;
+                scope.calculatedRatePerPeriod = total;
+
+                console.log(scope.formData);
```

Comment:
remove console log

Reply:
also remove extra new lines  used in different files

Reply:
Ok